### DISCUSSION TITLE
Language Switching

### DISCUSSION THREAD

**Discussion by jbeales** [ttscoff/bunch-docs#58] [Ideas]
I would like to be able to switch input languages with Bunch. For certain things, (like my French class), I want my keyboard input to be set to French, for others I want to make sure it's set to English.

Setting language looks like it's [kind of a pain with Applescript on recent versions of macOS](https://apple.stackexchange.com/questions/376922/switch-to-a-specific-input-source-in-os-x-in-catalina-with-applescript), but it looks like it might be doable, maybe even easier, in an actual program.

Let me know if you want an actual ticket opened up! This is the first time I've run across discussions in action here in Github.

**Comment by ttscoff** [comment]
I'll just continue this conversation here, we can just make a ticket out of it if something goes wrong :). 

I've implemented a rough draft of a command for the next update. It allows you to do something like:

```
(input source french)
```

Which will then find the first enabled input source matching "french". It also accepts a full identifier like `com.apple.keylayout.British` if you need to get more specific and avoid duplicate matches. It also records the current input source at the time it's changed, and can revert it when closing the Bunch that originally called it. This will probably need some smoothing out and I haven't tested it with multiple Bunches all changing the input source yet.

This command requires that the requested input source be installed and enabled. I don't think Bunch should deal with enabling/disabling sources, but this is a feature of macOS that I have little experience with. As someone who uses it, does the above sound sensible, or does it need to do more?

**Reply by jbeales** [reply]
Your fast! I didn't even bother making a nice ticket yet and you've got the feature working ;)

That sounds like exactly what I'm looking for. The languages I use are installed & enabled, I just want to switch automatically when I change from an English-based context to a French-based one. I'll test it out and let you know how it goes.

**Reply by ttscoff** [reply]
The new version is available now, did you get a chance to try out the new command? About ready to end the beta, but want to make sure I didn't introduce new bugs with this last-minute feature add.

**Reply by jbeales** [reply]
Hi - sorry for the delay. I just did some basic testing and it seems to work. I like that I don't have to match the input name exactly - for example, the French input I use is called "Canadian French - CSA" but (input source French) works fine. 

Now I've got the language set in a few of my bunches I'll let you know if anything goes wrong.
